staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/initializer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"context"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+type CELPolicyEvaluator interface {
+	Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error
+	HasSynced() bool
+}
+
+// Creates a plugin initializer which dependency injects a singleton
+// cel admission controller into the plugins which desire it
+func NewPluginInitializer(validator CELPolicyEvaluator) *PluginInitializer {
+	return &PluginInitializer{validator: validator}
+}
+
+// WantsCelAdmissionController gives the ability to have the shared
+// CEL Admission Controller dependency injected at initialization-time.
+type WantsCELPolicyEvaluator interface {
```

Comment:
There is the question of whether this initializer is best as a separate plugin initializer, or built into the generic apiserver initializer:

https://github.com/kubernetes/kubernetes/blob/f8211d7e447cc6c29139ebf3422f0752278d6da1/staging/src/k8s.io/apiserver/pkg/admission/initializer/initializer.go#L56-L78

Reply:
Is there extra benefit to have a separate plugin initializer?

Reply:
One drawback of generic plugin initializer is that all genericapiserver instances are expected to provide the admission controller if we add it there. 

https://github.com/kubernetes/kubernetes/blob/f8211d7e447cc6c29139ebf3422f0752278d6da1/staging/src/k8s.io/apiserver/pkg/admission/initializer/initializer.go#L35-L52

(I also feel like the cel admission controller would make an odd addition to the above argument list)

By using the separate initializer we can make it clearer that it is opt-in, if that is what we want.

FWIW Webhook also has a separate initializer used by `kubeapiserver` here:

https://github.com/kubernetes/kubernetes/blob/47ff6c6590932930f47b24e6c3769b98d0aa4d97/pkg/kubeapiserver/admission/config.go#L50-L52

Reply:
Strictly speaking, if we treat the exported code as an API, I believe making it opt-in is the backward compatible way of introducing this?  If so I'm in favor of that approach. I'm generally adverse to anything that could be called a breaking change for an alpha feature.